2_process/src/generate_diagnostics.R

```diff
@@ -0,0 +1,37 @@
+#' Generate diagnostic text based on model results
+#' 
+#' Takes model output, calculates summary statistics for models
+#' and produces a text file
+#' 
+#' @param data_in tbl, model results from ScienceBase
+#' @param file_out string, location where diagnostic text will be saved
+#' 
+generate_diagnostics <- function(data_in, file_out){
+  # Save the model diagnostics
+  render_data <- list(pgdl_980mean = filter(eval_data, model_type == 'pgdl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_980mean = filter(eval_data, model_type == 'dl', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_980mean = filter(eval_data, model_type == 'pb', exper_id == "similar_980") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_500mean = filter(eval_data, model_type == 'dl', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_500mean = filter(eval_data, model_type == 'pb', exper_id == "similar_500") %>% pull(rmse) %>% mean %>% round(2),
+                      dl_100mean = filter(eval_data, model_type == 'dl', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_100mean = filter(eval_data, model_type == 'pb', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
+                      pgdl_2mean = filter(eval_data, model_type == 'pgdl', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2),
+                      pb_2mean = filter(eval_data, model_type == 'pb', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2))
```

Comment:
You should make a function for every repeated expression. So here we see the same filter command applied to each of the datasets. The filter command should be a function. That way if we ever decide to change the filter, we just change the function once

Reply:
Great suggestion. I implemented it, but in a slightly different way. This because of this line in your comment:
> ...if we ever decide to change the filter, we just change the function once

On it's way to generating a the summary text file, the original `generate_diagnostics` function generates a named list with 9 elements: calculating summary stats for a manually specified list of model results in the `eval_data` dataset. 

My update does a few things. First, I added a function called `render_stats` to replace the multiple calls to `filter` that you reference above. Instead of calculating stats for a manually specified list of model results, `render_stats` calculates stats for all unique combinations of model and profiles. This results in a named list with 17 elements. the naming scheme between the two methods matches so that the whisker template still works.

Second, I pulled out a few elements that the user may want to change when changing the filter. I added a `FUN` argument that allows for the user to change the summary function from `mean` to whatever else they would like to use. I also pulled out the `digits` argument (called `rd` in t`render_stats`).

Finally, I manually verified that the txt output between the original implementation and the update match (I'm having some PR account issues that prevented an automated compare).

Reply:
Also, what is the etiquette around "resolving a conversation". Should I do it or should I let the reviewer decide?

Reply:
Looks great! What is the issue that not allowing you to compare the txt files? I think a `diff` should work, but is there a PR need to use `diff`?

For resolving comments, the reviewer usually is the one resolving them. Sometimes new commits outdate comments, so those are sort of "resolved" by the person committing instead